2022-11-23_proposal-process.md

```diff
@@ -0,0 +1,131 @@
+* Design docs do not only relate to Prometheus repo but full ecosystem or even neighbour projects like Alertmanager and clients. Ideally, we can share the same design process across all the things Prometheus Team and community help with.
+
+2. Different placement of design docs in markdown: `prometheus/docs`
+
+We could place them in [Prometheus docs](https://github.com/prometheus/docs), which already hosts [index of design docs](https://github.com/prometheus/docs/blob/main/content/docs/introduction/design-doc.md). There are some issues too:
+
+* Still, some CI checks for a website would be unrelated.
+* Design docs buried somewhere in website implementation and content, despite not rendering the proposals on the website. Less discoverable, and it also makes it harder to categorize GitHub Discussions and Issue related for design docs.
+
+3. Stick to Google Docs for design docs.
+
+It's worth exploring the idea of sticking to Google Docs as we do right now and trying to mitigate the flaws mentioned in [Pitfalls of the current solution](#pitfalls-of-the-current-solution) in some way.
+
+* We could ask to use a consistent template.
+* We could maintain Prometheus Google Drive and ask owners to transfer ownership to us, which would give some immutability. Still tracking consensus and maintaining this is non-trivial. Discovery is also poor and would mean extra work to maintain the index page like https://prometheus.io/docs/introduction/design-doc/
+
+4. Have `Accepted`, `Rejected`, and `Implemented` directories for different statuses
+
+This is a double maintenance effort. Using GitHub PR Close or Merge actions already indicates approval or rejection, so why not use that?
+
+The implementation status proved to be stale very quickly. Manual interactions to update it is not viable, as we see from the current index page.
+
+5. Move some metadata to the front matter.
+
+We could put the title, author, PRs, data and implementation status as YAML in [front matter](https://frontmatter.codes/docs/markdown#front-matter-highlighting) in markdown. This is useful if we would like to build further automation.
+
+Given YAGNI, I think it's not needed at this point. The regular markdown list is good enough and can be automated/changed in later iterations of the proposal repo.
+
+## Action Plan
+
+The tasks to do in order to migrate to the new idea:
+
+* [ ] Copy https://github.com/bwplotka/proposals to github.com/prometheus/proposals
+* [ ] Copy instructions from here to README.md
+* [ ] Migrate all the accepted docs from https://prometheus.io/docs/introduction/design-doc/ (mainly from Google Docs). Update status on the way (things are not up-to-date).
+* [ ] Migrate TODO ideas to GH issues.
+* [ ] Decide what to do with in-progress unapproved proposals. Potentially move to PRs (or ask owners to do so?)
```

Comment:
As a first step, we can make issues to "migrate Proposal X" with a link to the existing one. That way we can represent them quickly in the new place, without blocking on reformatting and/or the owner to pick it up.